src/libtrx/game/anims/frames.c

```diff
@@ -54,17 +61,36 @@ static int32_t M_ParseFrame(ANIM_FRAME *const frame, const int16_t *data_ptr)
     frame->offset.x = *data_ptr++;
     frame->offset.y = *data_ptr++;
     frame->offset.z = *data_ptr++;
-    frame->nmeshes = *data_ptr++;
+    #if TR_VERSION == 1
+    ASSERT(mesh_count == *data_ptr++);
+    #endif
 
     frame->mesh_rots =
-        GameBuf_Alloc(sizeof(int32_t) * frame->nmeshes, GBUF_ANIM_FRAMES);
-    memcpy(frame->mesh_rots, data_ptr, frame->nmeshes * sizeof(int32_t));
-    data_ptr += frame->nmeshes * sizeof(int32_t) / sizeof(int16_t);
+        GameBuf_Alloc(sizeof(XYZ_16) * mesh_count, GBUF_ANIM_FRAMES);
+    for (int32_t i = 0; i < mesh_count; i++) {
+        XYZ_16 *const rot = &frame->mesh_rots[i];
+        data_ptr += M_ParseMeshRotation(rot, data_ptr);
+    }
 
     return data_ptr - frame_start;
 #endif
 }
 
+static int32_t M_ParseMeshRotation(XYZ_16 *const rot, const int16_t *data_ptr)
+{
+#if TR_VERSION == 1
+    const int16_t rot_val_1 = *data_ptr++;
+    const int16_t rot_val_2 = *data_ptr++;
+    rot->x = EXTRACT_ROT_X(rot_val_2);
+    rot->y = EXTRACT_ROT_Y(rot_val_2, rot_val_1);
+    rot->z = EXTRACT_ROT_Z(rot_val_1);
+    return 2;
+#else
+    ASSERT_FAIL();
+    return 0;
+#endif
+}
```

Comment:
This function could take a double pointer removing the need for a return value.

Reply:
Done